### PR TITLE
Add cold boot option to emulator launch command

### LINKED ISSUE #56828: `flutter emulator --launch --coldboot` feature request
## Use case

Sometimes we encounter weird behavior with our AVD so that it just freezes and only thing helping is a cold boot. Unfortunately, to my knowledge, cold booting is possible only with AVD manager in Android Studio, but I develop in VS Code and launching heavy Android Studio just for that is really not convenient.

I would like to just have the possibility to run my device cold booted using Flutter CLI:

    flutter emulators --launch Pixel_2_XL_API_28 --coldboot

## Proposal

Add `--coldboot` (or `--cold`) parameter to `flutter emulators` tool that launches the emulator cold booted.

### PR DESCRIPTION
**Pull request by Hamdor**
The cold boot option is only available for AVD.

This PR implements the following feature request: #56828

closes #56828

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_emulator.dart [hunk 1]
```diff
@@ -149,10 +149,11 @@ class AndroidEmulator extends Emulator {
   String _prop(String name) => _properties != null ? _properties[name] : null;
 
   @override
-  Future<void> launch({@visibleForTesting Duration startupDuration}) async {
-    final Process process = await _processUtils.start(
-      <String>[_androidSdk.emulatorPath, '-avd', id],
-    );
+  Future<void> launch({@visibleForTesting Duration startupDuration, bool cold = false}) async {
+    final List<String> command = <String>[_androidSdk.emulatorPath, '-avd', id]
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_emulator.dart:153]: You can write this more compactly using an `if` in the list literal:

```dart
final List<String> cmd = <String>[
  _androidSdk.emulatorPath,
  '-avd',
  id,
  if (cold)
    '-no-snapshot-load',
];
```

File: packages/flutter_tools/test/general.shard/android/android_emulator_test.dart [hunk 2]
```diff
@@ -144,6 +144,20 @@ void main() {
       await emulator.launch(startupDuration: Duration.zero);
     });
 
+    testWithoutContext('succeeds with coldboot launch', () async {
+      final List<String> kEmulatorLauchColdBootCommand = kEmulatorLaunchCommand
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/android/android_emulator_test.dart:148]: nit: use a `...`:

```dart
final List<String> kEmulatorLauchColdBootCommand = <String>[...kEmulatorLaunchCommand, '-no-snapshot-load'];
```

File: packages/flutter_tools/lib/src/emulator.dart [hunk 3]
```diff
@@ -268,7 +268,7 @@ abstract class Emulator {
         && other.id == id;
   }
 
-  Future<void> launch();
+  Future<void> launch({bool cold});
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/emulator.dart:271]: I would name this `coldBoot`

**Comment by Hamdor** [comment]
@jonahwilliams, thank you for the review. I addressed your findings in the latest commit.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**